crates/uv-interpreter/src/interpreter.rs

```diff
@@ -324,13 +304,48 @@ pub(crate) struct InterpreterQueryResult {
 impl InterpreterQueryResult {
     /// Return the resolved [`InterpreterQueryResult`] for the given Python executable.
     pub(crate) fn query(interpreter: &Path) -> Result<Self, Error> {
-        let output = Command::new(interpreter)
-            .args(["-c", include_str!("get_interpreter_info.py")])
-            .output()
-            .map_err(|err| Error::PythonSubcommandLaunch {
-                interpreter: interpreter.to_path_buf(),
-                err,
-            })?;
+        let script = include_str!("get_interpreter_info.py");
+        let output = if cfg!(windows)
+            && interpreter
+                .extension()
+                .is_some_and(|extension| extension == "bat")
```

Comment:
Should it be `extension == "bat"` or `extension != "exe"`? 
Are there other extensions that could be used as shims? Maybe `.ps1`?

Reply:
For now, we only need to support installations that can be found by `find_python` and it only tests for `exe` (preferred) or `bat`). We could consider supporting `ps1` as well if there are existing shims that don't provide a batch equivalent. 

The reason I went with `bat` only for now is that testing additional extensions comes at a cost and I want to avoid paying that cost if limiting us to `bat` is sufficient.

Reply:
Not aware. Pyenv creates a shim without extension + a .bat shim.
![image](https://github.com/astral-sh/uv/assets/1867551/2efb0518-7935-431c-b016-e0951741e792)